pybamm/solvers/base_solver.py

```diff
@@ -797,10 +797,18 @@ def solve(
                 pybamm.SolverWarning,
             )
 
-        # Identify the event that caused termination
-        termination = self.get_termination_reason(solutions[0], model.events)
+        # Identify the event that caused termination and update the solution to
+        # include the event time and state
+        # TODO: when solving with a list of inputs each solution may terminate due
+        # to a different event - this isn't captured here (or in the logger info)
+        solutions[0], termination = self.get_termination_reason(
+            solutions[0], model.events
+        )
```

Comment:
isn't this a trivial for loop?

Reply:
Yeah, I’ll add it. For the logger info shall i just give different info when it’s a list of inputs (ie don’t specify a termination reason since it may be different for each solution).

Reply:
Yeah sounds good